src/compat/posix/posix_memalign.c

```diff
@@ -0,0 +1,61 @@
+/**
+ * @file
+ * @brief
+ *
+ * @author  KOUAM Ledoux
+ * @date    04.07.2020
+ */
+#include <errno.h>
+#include <unistd.h>
+#include <mem/vmem.h>
+#include <stdlib.h>
+/*
+    * The  function  posix_memalign() allocates size bytes and places the address of the allocated memory in *__memptr.  
+    * The address of the allocated memory will be a multiple of __alignment, which must be a power of two and a multiple of sizeof(void *).  
+    * If __size is 0, then the value placed in  *__memptr is either NULL, or a unique pointer value that can later be successfully passed to free().
+*/
+int power_of_two(int nmber) {
+    if ((nmber & (nmber - 1 )) == 0) {
+        return 0 ;
+    } else {
+        return 1 ;
+    }
+}
+int posix_memalign (void **__memptr, size_t __alignment, size_t __size) {
+    if (power_of_two(__alignment) != 0 || __size == 0){
+        return EINVAL;
+    }
+    *__memptr = memalign(__alignment,  __size) ;
```

Comment:
Why you are using such `__`?

Reply:
i have looked from the the linux manual , sorry i will remove it